Production Option Value Translations could not be getting missed now

MarekB01 opened the pull request:
https://github.com/BroadleafCommerce/QA/issues/4906

Added a client's code into our codebase, added logger

MarekB01 committed:
BroadleafCommerce/QA-4906_Production_Option_Value_Translations_could_be_getting_missed

Added a client's code into our codebase, added logger

oleksiimiroshnyk commented on core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/ProductOptionDisplayProcessor.java:
```diff
@@ -40,48 +44,78 @@
 @ConditionalOnTemplating
 public class ProductOptionDisplayProcessor extends AbstractBroadleafVariableModifierProcessor {
 
+    private static final Logger logger = LoggerFactory.getLogger(ProductOptionDisplayProcessor.class);
```
better do it protected

oleksiimiroshnyk commented on core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/ProductOptionDisplayProcessor.java:
```diff
@@ -40,48 +44,78 @@
-                    ) {
-                        productOptionDisplayValues.put(label, orderItem.getOrderItemAttributes().get(i).toString());
+            // Do a search only on ProductOptionXrefs, and find the translation by matching user's input to one of the ProductOptionValues present on ProductOption.getAllowedValues().
+            // Here, in our customisation, the ProductOptionXrefs are searched in a specific order (by also using an additional attribute "sequence" on FlywireProductOptionXref)
```
```suggestion

```

oleksiimiroshnyk commented on core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/ProductOptionDisplayProcessor.java:
```diff
@@ -40,48 +44,78 @@
-                    String optionValue = productOptionValueXref.getProductOptionValue().getAttributeValue();
-                    productOptionDisplayValues.put(label, optionValue);
-                }
+            // The search of ProductOptions on Sku was removed because we cannot properly sort the items based on their displayOrder / sequence
```
```suggestion
```

oleksiimiroshnyk commented on core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/ProductOptionDisplayProcessor.java:
```diff
@@ -40,48 +44,78 @@
-        Object item = context.parseExpression(tagAttributes.get("orderItem"));
+    public Map<String, Object> populateModelVariables(final String tagName, final Map<String, String> tagAttributes, final BroadleafTemplateContext context) {
+
+        // on our side, it is important to keep the order in which the product options should be displayed
```
```suggestion

```

oleksiimiroshnyk commented on core/broadleaf-framework-web/src/main/java/org/broadleafcommerce/core/web/processor/ProductOptionDisplayProcessor.java:
```diff
@@ -40,48 +44,78 @@
+                final ProductOption productOption = productOptionXref.getProductOption();
+                final OrderItemAttribute itemAttribute = orderItemAttributes.get(productOption.getAttributeName());
+
+                // Translate and collect only those options for which a user input was selected (on orderItem) - on current BL implementation, empty options are still collected in the result
```
```suggestion
                // Translate and collect only those options for which a user input was selected (on orderItem)
```

MarekB01 committed:
BroadleafCommerce/QA-4906_Production_Option_Value_Translations_could_be_getting_missed

Added a client's code into our codebase, added logger. Refactored code, removed comments

Commit:
BroadleafCommerce/QA#4906
- changed private to protected
